### PR TITLE
Newplant

### PR DESCRIPTION
**Pull request by bomentum**
1) npm install react-select-search for search bar. should return searchable dropdown list of species
2) need to still post to plant db
3) problems with that backend/index.js code when trying to post/get only for plants db. anyone else?

### CODE REVIEW

**Comment by bomentum** [comment]
ignore earnest-bienenstitch-3e6bb8 
Jarod's meek-souffle-992b4e is the group's deploy site

File: src/modules/Data.js [hunk 1]
```diff
@@ -1,88 +1,97 @@
...
+        method: 'GET',
+        headers: {
+            Authorization: `Bearer: ${token}`,
+        },
+    });
+    if (!result.ok) return -1;
+    return await result.json();
+}
+
+export async function getReqSpecies(species, token) {
```
Comment (Jrendl) [inline src/modules/Data.js]: can u find a better name for this than ``getReqSpecies``? not very descriptive. maybe ``getSpeciesbyName`` or something

File: src/modules/Data.js [hunk 2]
```diff
@@ -1,88 +1,97 @@
...
+            Authorization: `Bearer: ${token}`,
+        },
+    });
+    if (!result.ok) return result.response;
+    return await result.json();
 }
 
 export async function postSpecies(speciesJSON, token) {
     const result = await fetch(`${backend_base}/species/`, {
-        'method': 'POST',
```
Comment (Jrendl) [inline src/modules/Data.js:17]: I'm confused where these changes r coming from ngl. what did u purposefully change in this doc?

File: src/pages/newplant.js [hunk 3]
```diff
@@ -50,54 +66,59 @@ export default function NewPlant() {
...
                     </div>
+                    <div class='field'>
+                        <div class='label'>Add image component here</div>
+                        <figure class='image is-128x128'>
+                            <img src='/sun.png'></img>
+                        </figure>
+                    </div>
+
+                    <div class='control'>
+                        <button class='button is-large'>Add Plant</button>
```
Comment (Jrendl) [inline src/pages/newplant.js]: you never actually submit the form from what I can tell so it doesn't actually add the plant to the db. When I go back to the home page the plant that I added doesn't appear

**Review by Jrendl — CHANGES_REQUESTED** [review]
A few things and a bug. Also gonna need to rebase since I merged recently

**Comment by Jrendl** [comment]
would also be nice to search species by common name not species name

File: src/modules/Data.js (hunk 1 above)
Comment (bomentum) [reply src/modules/Data.js]: getSpeciesByName done

File: src/modules/Data.js (hunk 2 above)
Comment (bomentum) [reply src/modules/Data.js:17]: i think these changes were from Jarod's merge from last week. i just added getReqSpecies.

File: src/pages/newplant.js (hunk 3 above)
Comment (bomentum) [reply src/pages/newplant.js]: adding new plant is mostly done, waiting for how we're going to do images

File: src/modules/Data.js (hunk 2 above)
Comment (bomentum) [reply src/modules/Data.js:17]: oh, i see, it was the prettier formatting that auto added commas and semicolons

**Review by Jrendl — APPROVED** [review]

**Pull request merged**